analyses/add_leaflet_maps.qmd

```diff
@@ -0,0 +1,226 @@
+```{r}
+create_leaflet_map <- function(dataset, legend_value, legend_title, order_scheme = "high", longitude = "loc_longitude", latitude = "loc_latitude") {
+  # Filter neighborhoods that have at least one observation
+  nbhd_borders <- nbhd %>%
+    right_join(dataset, by = c("town_nbhd" = "meta_nbhd_code"))
+
+  # Create the color palette based on order_scheme
+  if (order_scheme == "low") {
+    pal <- colorNumeric(palette = "Reds", domain = dataset[[legend_value]], reverse = TRUE)
+  } else {
+    pal <- colorNumeric(palette = "Reds", domain = dataset[[legend_value]])
+  }
+
+  # Calculate the bounding box of the filtered neighborhoods
+  bbox <- st_bbox(nbhd_borders)
+
+  # Create the leaflet map
+  leaflet(dataset) %>%
+    addProviderTiles(providers$CartoDB.Positron) %>%
+    addCircleMarkers(
+      lng = ~ get(longitude),
+      lat = ~ get(latitude),
+      radius = 5,
+      color = ~ pal(dataset[[legend_value]]),
+      popup = ~ {
+        shap_values <- dataset %>%
+          select(starts_with("target_feature_shap_")) %>%
+          summarise_all(~ ifelse(!is.na(.), sprintf("SHAP: %s", scales::dollar(.)), NA)) %>%
+          apply(1, function(row) {
+            paste(na.omit(row), collapse = "<br>")
+          })
+        paste(
+          "Pin: ", meta_pin,
+          ifelse(shap_values == "", "", paste0("<br>", shap_values)),
+          "<br>", "Relative SHAP: ", scales::percent(relative_shap, accuracy = 0.01),
+          "<br>", "Feature: ", sprintf("%.2f", get(params$added_feature)),
+          "<br>", "New FMV: ", pred_pin_final_fmv_new,
+          "<br>", "Comparison FMV: ", pred_pin_final_fmv_comp,
+          "<br>", "FMV Difference: ", scales::percent(diff_pred_pin_final_fmv)
+        )
+      }
+    ) %>%
+    addPolygons(
+      data = nbhd_borders,
+      color = "black",
+      weight = 2,
+      fill = FALSE
+    ) %>%
+    addLegend(
+      "bottomright",
+      pal = pal,
+      values = dataset[[legend_value]],
+      title = legend_title
+    )
+}
+```
+
+## Highest and Lowest 100 Values
```

Comment:
This is the only map which doesn't work with categorical, and I don't see a readily comparable option.